### PR TITLE
Rename Config Mixins to Fragments

### PR DESCRIPTION
**Pull request by abejgonzalez**
**Related issue**: 

**Type of change**: other enhancement

**Impact**: other

**Release Notes**
This renames all instances of config mixins to fragments to better distinguish between traits/mixins that apply to the `Top` and config fragments that make configs.

### CODE REVIEW

**Comment by alonamid** [comment]
I'd rather we use the full term "config fragment" every time we refer to it, rather than just "fragment". I think it maker it clearer, and keeps it within the context of configs

File: docs/Generators/IceNet.rst [hunk 1]
```diff
@@ -80,8 +80,8 @@ and ``HasPeripheryIceNICModuleImp`` to the module implementation. If you
 are confused about the distinction between lazy module and module
 implementation, refer to :ref:`Cake Pattern`.
 
-Then add the ``WithIceNIC`` config mixin to your configuration. This will
-define ``NICKey``, which IceNIC uses to determine its parameters. The mixin
+Then add the ``WithIceNIC`` config fragment to your configuration. This will
+define ``NICKey``, which IceNIC uses to determine its parameters. The fragment
```
Comment (abejgonzalez) [inline docs/Generators/IceNet.rst]: Address this.

File: generators/chipyard/src/main/scala/GCD.scala [hunk 2]
```diff
@@ -200,8 +200,8 @@ trait CanHavePeripheryGCDModuleImp extends LazyModuleImp {
 // DOC include end: GCD imp trait
 
 
-// DOC include start: GCD mixin
+// DOC include start: GCD fragment
```
Comment (abejgonzalez) [inline generators/chipyard/src/main/scala/GCD.scala]: Address this.

File: generators/chipyard/src/main/scala/ConfigFragments.scala [hunk 3]
```diff
@@ -31,21 +31,21 @@ object ConfigValName {
...
+// Common Parameter Fragments
+// --------------------------
 
 class WithBootROM extends Config((site, here, up) => {
   case BootROMParams => BootROMParams(
     contentFileName = s"./bootrom/bootrom.rv${site(XLen)}.img")
 })
 
-// DOC include start: gpio mixin
+// DOC include start: gpio fragment
```
Comment (abejgonzalez) [inline generators/chipyard/src/main/scala/ConfigFragments.scala]: Address this.

File: docs/Generators/IceNet.rst (hunk 1 above)
Comment (alonamid) [inline docs/Generators/IceNet.rst]:
```suggestion
define ``NICKey``, which IceNIC uses to determine its parameters. The config fragment
```

**Pull request merged**